src/main/java/mediator/ConcreteMediator.java

```diff
@@ -49,7 +49,7 @@ private ConcreteMediator() {
      *
      * @return The Singleton instance of the ConcreteMediator class.
      */
-    public static ConcreteMediator getInstance() {
+    public static synchronized ConcreteMediator getInstance() {

```

Comment:
This will be getting updated in a separate commit. Basically exactly what I mentioned above for CarrierManagementSystemDirector.

Reply:
In that case I'll take this out and leave it for the separate commit from you to come in. I'd still like to discuss whether we should go forward with double checked locking over eager instantiation at some point though.